File: solutions/cpp/leap/1/leap.cpp
#include "leap.h"

namespace leap {

bool is_leap_year(int currentYear) {
    bool isLeapYear{false};
    if (currentYear % 100 == 0) {
        if (currentYear % 400 == 0) {
            isLeapYear = true;
        } else {
            isLeapYear = false;
        }
    } else if (currentYear % 4 == 0){
        isLeapYear = true;
    } else {
        isLeapYear = false;
    }

    return isLeapYear;
}

}  // namespace leap

File: solutions/go/leap/1/leap.go
// Package leap should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package leap

// IsLeapYear determines if an input year is a leap year.
func IsLeapYear(year int) bool {
	if year%100 == 0 {
		if year%400 == 0 {
			return true
		}
	} else if year%4 == 0 {
		return true
	}

	return false
}
